tests/test_enterprise/views/test_grant_data_sharing_permissions.py

```diff
@@ -640,28 +681,33 @@ def test_post_program_consent_missing_parameter(self, missing_parameter):
         valid_post_params = self.valid_post_params.copy()
         valid_post_params.pop(missing_parameter)
         self._login()
+        self._assert_post_returns_404_with_mock(self.url, valid_post_params)
+
+    def test_post_program_consent_missing_parameter_enterprise_customer(self):
+        self.course_catalog_api_client.return_value.program_exists.return_value = True
+        valid_post_params = self.valid_post_params.copy()
+        valid_post_params.pop('enterprise_customer_uuid')
+        self._login()
         response = self.client.post(self.url, valid_post_params)
         assert response.status_code == 404
 
     def test_post_consent_program_does_not_exist(self):
         self.course_catalog_api_client.return_value.program_exists.return_value = False
         self._login()
-        response = self.client.post(self.url, self.valid_post_params)
-        assert response.status_code == 404
+        self._assert_post_returns_404_with_mock(self.url, self.valid_post_params)
 
     def test_post_program_consent_no_ec(self):
         self.get_data_sharing_consent.return_value = None
         self.course_catalog_api_client.return_value.program_exists.return_value = True
         self._login()
-        response = self.client.post(self.url, self.valid_post_params)
-        assert response.status_code == 404
+        self._assert_post_returns_404_with_mock(self.url, self.valid_post_params)
 
     def test_post_program_consent_not_required(self):
         self.get_data_sharing_consent.return_value.consent_required.return_value = False
         self.course_catalog_api_client.return_value.program_exists.return_value = True
         self._login()
-        response = self.client.get(self.url, self.valid_post_params)
-        assert response.status_code == 404
+        response = self.client.post(self.url, self.valid_post_params)
+        assert response.status_code == 302
```

Comment:
why is this a 302 now?

Reply:
the test name was test_post_program_consent_not_required but the call in the test was a get. i changed it to a post, but a post with the same parameters results in a redirect, and not a 404, so i updated the result as well